script.js

```diff
@@ -1,4 +1,318 @@
+        outputMsg += `${allPlayerCard[i].name}'s card total is ${allPlayerCard[i].bestTotal}, ${allPlayerCard[i].name} lost! <br>`;
+      }
+    }
+  }
+  return outputMsg;
+};
+
+var decidePlayerBestTotal = function () {
+  for (var i = 0; i < allPlayerCard.length; i += 1) {
+    var firstPossibleTotal = allPlayerCard[i].cardTotal[0];
+    var secPossibleTotal = allPlayerCard[i].cardTotal[1];
+    if (firstPossibleTotal > secPossibleTotal && firstPossibleTotal <= 21) {
+      allPlayerCard[i].bestTotal = firstPossibleTotal;
+    } else allPlayerCard[i].bestTotal = secPossibleTotal;
+  }
+};
+
+var resetGame = function () {
+  allPlayerCard = [];
+  cardDeck = makeDeck();
+  currentPlayer = 0;
+  numberOfPlayer = 0;
+  gameMode = "A";
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var outputMsg = "";
+  if (gameMode == "A" && currentPlayer < numberOfPlayer) {
+    outputMsg = checkInitialWinning();
+    if (input == "hit") {
+      if (
+        allPlayerCard[currentPlayer].cardTotal[0] > 21 &&
+        allPlayerCard[currentPlayer].cardTotal[1] > 21
+      ) {
+        outputMsg = `You have exceeded maximum 21. Please click <b>stay</b> to next player!`;
+      } else var playerNumberOfCards = allPlayerCard[currentPlayer].card.length;
+      allPlayerCard[currentPlayer].card[playerNumberOfCards] =
+        shuffledDeck.pop();
+      updateNumberOfAce();
+    }
+    if (input == "stay") {
+      if (
+        allPlayerCard[currentPlayer].cardTotal[0] > 15 ||
+        allPlayerCard[currentPlayer].cardTotal[1] > 15 ||
+        allPlayerCard[currentPlayer].blackJack == true
+      ) {
+        currentPlayer += 1;
+      } else
+        outputMsg = `Your card total did not meet minimum 15. <br> Please click <b>hit</b> to draw one more card!`;
+    }
+    if (outputMsg == "") {
+      updateCardTotal();
+      outputMsg = displayCardsDrawn();
+      if (
+        allPlayerCard[currentPlayer].card.length == 5 &&
+        allPlayerCard[currentPlayer].cardTotal[0] <= 21
+      ) {
+        outputMsg += ` ${allPlayerCard[currentPlayer].name} won! <br>`;
+        if (allPlayerCard.length == 2) {
+          hit.disabled = true;
+          stay.disabled = true;
+          reset.style.visibility = "visible";
+        } else outputMsg += `Please click <b>stay</b> to move to next player.`
+      }
+      if (
+        allPlayerCard[currentPlayer].card.length == 5 &&
+        allPlayerCard[currentPlayer].cardTotal[0] > 21
+      ) {
+        outputMsg += ` ${allPlayerCard[currentPlayer].name} lost! <br>`;
+        if (allPlayerCard.length == 2) {
+          hit.disabled = true;
+          stay.disabled = true;
+          reset.style.visibility = "visible";
+        } else outputMsg += `Please click <b>stay</b> to move to next player.`
+      }
+    }
+    if (currentPlayer == numberOfPlayer) {
+      updateCardTotal();
+      checkDealerCard();
+      outputMsg = displayCardsDrawn();
+      gameMode = "B";
```

Comment:
Same as above comment, `B` doesn't explain what the game mode is, need to choose a better name